pkg/controller/controlplane/valuesprovider.go

```diff
@@ -668,12 +673,128 @@ func getCSIControllerChartValues(
 			},
 		},
 	}
+
+	if csi != nil {
+		if csi.CSIAttacher != nil {
+			values["csiAttacher"] = getCSIAttacherArgs(csi.CSIAttacher)
+		}
+		if csi.CSIResizer != nil {
+			values["csiResizer"] = getCSIResizerArgs(csi.CSIResizer)
+		}
+		if csi.CSIDriverCinder != nil {
+			values["csiDriverCinder"] = getCSIDriverCinderArgs(csi.CSIDriverCinder)
+		}
+		if csi.CSISnapshotController != nil {
+			values["csiSnapshotController"] = getCSISnapshotControllerArgs(csi.CSISnapshotController)
+		}
+		if csi.CSILivenessProbe != nil {
+			values["csiLivenessProbe"] = getCSILivenessProbeArgs(csi.CSILivenessProbe)
+		}
+		if csi.CSIProvisioner != nil {
+			values["csiProvisioner"] = getCSIProvisionerArgs(csi.CSIProvisioner)
+		}
+		if csi.CSISnapshotter != nil {
+			values["csiSnapshotter"] = getCSISnapshotterArgs(csi.CSISnapshotter)
+		}
```

Comment:
If we adapt the `CSI` struct to use the `CSIBaseArgs` for those field we can also just use the `getCSIBaseArgs()` function with different inputs. Only the `CSIAttacher` needs its own `get...Args()` func and can call internally `getCSIBaseArgs()`.

Reply:
Alright I tried to shrink the code as much as possible.